radix-engine/src/system/system_callback.rs

```diff
@@ -113,42 +150,110 @@ pub struct System<C: SystemCallbackObject> {
 impl<C: SystemCallbackObject> KernelCallbackObject for System<C> {
     type LockData = SystemLockData;
     type CallFrameData = Actor;
-    type InitInput = C::InitInput;
+    type InitInput = SystemInit<C::InitInput>;
 
     fn init<S: BootStore>(
         store: &S,
         executable: &Executable,
-        execution_config: &ExecutionConfig,
-        init_input: C::InitInput,
+        init_input: SystemInit<C::InitInput>,
     ) -> Result<Self, BootloadingError> {
-        let costing_parameters = {
+        let mut system_parameters = {
             let system_boot = store
                 .read_substate(
                     TRANSACTION_TRACKER.as_node_id(),
                     BOOT_LOADER_PARTITION,
                     &SubstateKey::Field(BOOT_LOADER_SYSTEM_SUBSTATE_FIELD_KEY),
                 )
                 .map(|v| scrypto_decode(v.as_slice()).unwrap())
-                .unwrap_or(SystemBoot::V1 {
-                    costing_parameters: CostingParameters::default(),
-                });
+                .unwrap_or(SystemBoot::V1(SystemParameters::default()));
 
             match system_boot {
-                SystemBoot::V1 { costing_parameters } => costing_parameters,
+                SystemBoot::V1(system_parameters) => system_parameters,
             }
         };
-        let callback = C::init(store, init_input)?;
 
-        let mut modules = SystemModuleMixer::new(
-            execution_config.enabled_modules,
-            execution_config.network_definition.clone(),
+        let callback = C::init(store, init_input.callback_init)?;
+
+        let enabled_modules = {
+            let mut enabled_modules = EnabledModules::AUTH | EnabledModules::TRANSACTION_RUNTIME;
+            if !executable.is_system() {
```

Comment:
Nice - having this on the executable actually allows us to simply things in the node quite a lot I think.

Although can I suggest we add a Rust doc to `is_system()` to explain what it does? (e.g. doesn't enable limits and costing)

Or even more clearly - what do you think regarding having explicit `enable_transaction_costing() -> bool` and `enable_transaction_limits() -> bool` methods on `Executable`?

Reply:
Rust doc sounds like a good idea!

I don't like having explicit enabling in the `Executable` as an `Executable` should describe just the transaction we are trying to execute. The enabling of costing/limits is a system override configuration.